fennel/datasets/datasets.py

```diff
@@ -1030,15 +1037,32 @@ def make_types_optional(types: Dict[str, Type]) -> Dict[str, Type]:
         if self.how == "left":
             right_value_schema = make_types_optional(right_value_schema)
 
-        # Add right value columns to left schema. Check for column name collisions
+        # TODO(sat): Are the same checks here and in SchemaValidator redundant?
+        # If right_fields is set, check that it contains elements from right schema values and timestamp only
+        if self.right_fields is not None and len(self.right_fields) > 0:
```

Comment:
prefer to raise error in SchemaValidator as we can give a lot more context about the error there